README.md

```diff
@@ -38,9 +38,13 @@ func main() {
 	}
 
 	// start
-	err = c.Scan(context.TODO(), func(r *consumer.Record) bool {
+	err = c.Scan(context.TODO(), func(r *consumer.Record) consumer.ScanError {
 		fmt.Println(string(r.Data))
-		return true // continue scanning
+        // continue scanning
```

Comment:
wondering if we could use `return nil` for the success case. so the function signature is 

```
c.Scan(context.TODO(), func(r *consumer.Record) error {
```

then in the client loop we'd need to try to cast the error as `consumer.ScanError` if an error is returned

Reply:
This is to address  https://github.com/harlow/kinesis-consumer/issues/43. To finish off the otehr PR that is kind of hanging. 
It looks like it mentions about supporting  when a consumer will want to not checkpoint, but continue scanning the shard, and when a consumer also want to checkpoint also .. Unsure if returning 1 state will suffice.  Also Scan is a convenience wrapper around ScanShard.
What do u think?

Reply:
maybe thread regular `error` back up to the `err` case?